### ISSUE TITLE
Commit a1e3a7 breaks previously valid rules

### ISSUE THREAD

**Issue by levand** [rbrush/clara-rules#96]
This used to work correctly. As of a1e3a729b4af15a1608c9ce05cbbc98dafa10b20, it fails with a (seemingly spurious) error message.

```clojure
{:name "clara.test-destructured-binding/test-destructured-binding"
 :lhs [{:args [[e a v]]
        :type :foo
        :constraints [(= e 1) (= v ?value)]}]
 :rhs (println "The value was" ?value)}
```

Yields:

```
clojure.lang.ExceptionInfo: Using variable that is not previously bound. This can happen when a test expression uses a previously unbound variable, or if a a variable is referenced in a nested part of a parent expression, such as (or (= ?my-expression my-field) ...). Unbound variables: #{?value}
```

I created a branch in my own fork with a failing test case, at https://github.com/levand/clara-rules/tree/regression-a1e3a7

**Comment by levand** [comment]
Update:

It works if I swap the *order of the arguments to `=`*. `(= v ?value)` fails, while `(= ?value v)` works.

I strongly contend that this is still a bug, and possibly a worse one than before in that it subtly breaks every expectation Clojure sets up about expression evaluation.

**Comment by rbrush** [comment]
Agreed, it's a bug. I'm pretty sure I know what caused it and will get in a fix shortly.

**Comment by rbrush** [comment]
It should be fixed in the above commit. All of the other tests put the ?variable name first, so this class of issue was never exercised elsewhere. It is now, so we shouldn't run into this moving forward.

Thanks for reporting, and apologies for the regression!

**Comment by levand** [comment]
Awesome, thanks for the fantastic response time.

**Comment by rbrush** [comment]
Closing this since it appears to be working. It will be included in the 0.8.6 release, so the regression this ran into will be fixed without making it into a release.
